connection.go

```diff
@@ -29,30 +29,40 @@ type StreamHandler func(stream *Stream)
 type AuthHandler func(header http.Header, slot uint8, parent uint32) bool
 
 type idleAwareFramer struct {
-	f       *spdy.Framer
-	conn    *Connection
-	expired *time.Timer
-	startCh chan struct{}
+	f              *spdy.Framer
+	conn           *Connection
+	expired        *time.Timer
+	resetChan      chan struct{}
+	setTimeoutChan chan time.Duration
+	timeout        time.Duration
 }
 
 func newIdleAwareFramer(framer *spdy.Framer) *idleAwareFramer {
 	iaf := &idleAwareFramer{
-		f:       framer,
-		startCh: make(chan struct{}, 1),
+		f:              framer,
+		resetChan:      make(chan struct{}, 2),
+		setTimeoutChan: make(chan time.Duration),
+		expired:        time.NewTimer(0),
 	}
-	go iaf.monitor()
+	iaf.expired.Stop()
```

Comment:
Not sure I understand what is stopping the timer from putting a value on iaf.expired.C before this is called. Stop does not seem to clear the channel of any value waiting to be received.